### PR TITLE
use protobuf for client/api communication

### PR DESCRIPTION
**Pull request by bparees**

### CODE REVIEW

**Comment by smarterclayton** [comment]
Looks correct but we should double check the metrics on the run

File: pkg/cmd/dockerregistry/dockerregistry.go [hunk 1]
```diff
@@ -179,7 +179,10 @@ func Execute(configFile io.Reader) {
 func NewServer(ctx context.Context, dockerConfig *configuration.Configuration, extraConfig *registryconfig.Configuration) (*http.Server, error) {
 	setDefaultLogParameters(dockerConfig)
 
-	registryClient := client.NewRegistryClient(clientcmd.NewConfig().BindToFile(extraConfig.KubeConfig))
+	clientConfig := clientcmd.NewConfig()
+	clientConfig.CommonConfig.AcceptContentTypes = "application/vnd.kubernetes.protobuf,application/json"
+	clientConfig.CommonConfig.ContentType = "application/vnd.kubernetes.protobuf"
+	registryClient := client.NewRegistryClient(clientConfig.BindToFile(extraConfig.KubeConfig))
```
Comment (smarterclayton) [inline pkg/cmd/dockerregistry/dockerregistry.go]: BindToFile ignore whatever the original client config is, it looks like.

Comment (smarterclayton) [reply pkg/cmd/dockerregistry/dockerregistry.go]: It doesn't ignore it, but this is not working the way I would expect.  Still seeing json to the server in the metrics capture.

Comment (bparees) [reply pkg/cmd/dockerregistry/dockerregistry.go]: @smarterclayton hm. i just refactored it to add the content setting after the bindtofile call and pushed, but it sounds like you don't think that'll matter.

**Comment by bparees** [comment]
@smarterclayton do you want to pick this up again?

**Comment by smarterclayton** [comment]
Yeah I do

**Comment by bparees** [comment]
> Yeah I do

@smarterclayton ok, we'll have to chat about what is missing here then since you indicated this didn't accomplish the goal as i recall.

unless you just want to merge this and see what happens, which i'm ok with too.

/retest

**Comment by bparees** [comment]
ref from @mfojtik for whenever this gets picked up again:
https://github.com/openshift/library-go/blob/69ec53fc354e648d1b1bd08556e702639f494688/pkg/operator/staticpod/certsyncpod/certsync_cmd.go#L107

**Pull request closed without merging**